### PR TITLE
Fixes #32621 - Update useEffect to useDeepCompareEffect

### PR DESCRIPTION
**Pull request by parthaa**

### CODE REVIEW

**Review by jeremylenz — CHANGES_REQUESTED** [review]
Tested and everything seems to be working fine :)

The request changes is just so we can move the dependency to `@theforeman/vendor-core`
LGTM otherwise :+1:

File: webpack/scenes/ContentViews/Details/Histories/ContentViewHistories.js [hunk 1]
```diff
@@ -102,7 +103,7 @@ const ContentViewHistories = ({ cvId }) => {
       const newRows = buildRows(results);
       setRows(newRows);
     }
-  }, [JSON.stringify(response)], setMetadata, loading, setRows);
+  }, [response], setMetadata, loading, setRows);
```
[the remark this answers is not in the document]
Comment (jeremylenz) [reply webpack/scenes/ContentViews/Details/Histories/ContentViewHistories.js:106]: @parthaa this one still needs fixing :)

**Review by jeremylenz — APPROVED** [review]
APJ

**Comment by sjha4** [comment]
We can merge this and let the foreman-js PR get comments and we can use that if it's accepted. 👍🏽

**Comment by jeremylenz** [comment]
@parthaa looks like there's 1 test failure, maybe it needs a rebase?

**Comment by ekohl** [comment]
This broke packaging. https://github.com/theforeman/foreman-packaging/pull/6844 was opened to add it, but it's not clear to me if this should be in foreman-js or not.

**Pull request merged**